src/app/shared/data-storage.service.ts

```diff
@@ -0,0 +1,37 @@
+import { HttpClient } from "@angular/common/http";
+import { Injectable } from "@angular/core";
+import { Owner } from "../owner/owner.model";
+import { Pet } from "../pets/pet.model";
+@Injectable({ providedIn: 'root' })
+export class DataStorageService {
+    constructor(private http: HttpClient) { }
+    fetchOwners() {
+        return this.http
+            .get<Owner[]>(
+                'http://localhost:8080/owners'
+            )
+    }
+
+    storeOwner(owner: Owner) {
+        return this.http.post<Owner>('http://localhost:8080/owner', owner);
```

Comment:
this string is being used many many times, should move it to a constant and reuse it
http://localhost:8080